src/CommandingOfficers/Commander.java

```diff
@@ -463,6 +468,42 @@ private void readObject(ObjectInputStream stream) throws IOException, ClassNotFo
     aiController = AILibrary.getAIList().get(stream.readInt()).create(this);
   }
 
+  /** Return true if this Commander is password-protected, false else. */
+  public boolean hasPassword()
+  {
+    return password != null;
+  }
+
+  /** Salt the provided string with the Commander's salt and return the result. */
+  private UUID hashPass(String pass)
+  {
+    int numBytes = 8 + (2*pass.length());
+    ByteBuffer bb = ByteBuffer.allocate(numBytes);
+    bb.putLong(passSalt).put(pass.getBytes());
+    UUID hashedPass = UUID.nameUUIDFromBytes(bb.array());
```

Comment:
This apparently uses MD5, so I'd complain if these were *user* passwords.
But they're not, so whatever.
(https://stackoverflow.com/questions/29059530/is-there-any-way-to-generate-the-same-uuid-from-a-string)

Reply:
Hm. That's a cool utility, almost might be worth using just to have it around. Though if we really want to turn up the security, the right answer might be to add the PGP option later and do it proper.